OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -1346,17 +1202,194 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
+                    name match {
+                        case `addDataAuthority` ⇒ dataAuthorities += (parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "") + parameter.last.asVar.value.toString.
+                            replaceAll(paramRegex, ""))
+                        case `addDataPath` ⇒ dataPaths += (parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "") + parameter.last.asVar.value.toString.
+                            replaceAll(paramRegex, ""))
+                        case `addDataScheme` ⇒ dataSchemes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataSSP` ⇒ dataSSPs += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataType` ⇒ dataTypes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                    }
+                }
+            }
+            if (stmt.isAssignment) {
+                if (stmt.asAssignment.expr.isVirtualFunctionCall && stmt.asAssignment.expr.asVirtualFunctionCall.name == registerReceiver) {
+                    val virtualCall = stmt.asAssignment.expr.asVirtualFunctionCall
+                    if (virtualCall.receiver.asVar.definedBy.head > -1)
+                        registeredReceivers += ObjectType(statements(virtualCall.
+                            receiver.asVar.definedBy.head).asAssignment.expr.asGetField.objRef.asVar.value.toString.
+                            replaceAll(receiverRegex, "").replaceAll("\\.", "/"))
+                }
+            }
+        }
+    }
+}
+
+object ManifestParsingKey extends ProjectInformationKey[(Map[String, ListBuffer[ClassFile]], ListBuffer[IntentFilter]), Elem] {
+
+    override def requirements(project: SomeProject): ProjectInformationKeys = Nil
+
+    override def compute(project: SomeProject): (Map[String, ListBuffer[ClassFile]], ListBuffer[IntentFilter]) = {
+        val manifest = project.getProjectInformationKeyInitializationData(ManifestParsingKey)
+        if (manifest.isDefined) {
+            parseManifest(project, manifest.get)
+        } else null
+    }
+
+    /**
+     * Analyses the AndroidManifest.xml of the project to find intent filters and relevant components to generate
+     * lifecycle callbacks.
+     */
+    def parseManifest(project: SomeProject, manifest: Elem): (Map[String, ListBuffer[ClassFile]], ListBuffer[IntentFilter]) = {
+        val componentMap: mutable.Map[String, ListBuffer[ClassFile]] = mutable.Map(
+            "activity" -> ListBuffer.empty[ClassFile],
+            "service" -> ListBuffer.empty[ClassFile]
+        )
+        val intentFilters = ListBuffer.empty[IntentFilter]
+        val androidURI = "http://schemas.android.com/apk/res/android"
+        val pack = manifest.attribute("package").get.toString().replaceAll("\\.", "/")
```

Comment:
Avoid using short names, e.g., `packageName` would probably be much clearer here.